src/qibolab/instruments/qblox/acquisition.py

```diff
@@ -0,0 +1,144 @@
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+import numpy as np
+
+SAMPLING_RATE = 1e9
+
+
+def demodulate(input_i, input_q, frequency):
+    """Demodulates and integrates the acquired pulse."""
+    # DOWN Conversion
+    # qblox does not remove the offsets in hardware
+    modulated_i = input_i - np.mean(input_i)
+    modulated_q = input_q - np.mean(input_q)
+
+    num_samples = modulated_i.shape[0]
+    time = np.arange(num_samples) / SAMPLING_RATE
+    cosalpha = np.cos(2 * np.pi * frequency * time)
+    sinalpha = np.sin(2 * np.pi * frequency * time)
+    demod_matrix = np.sqrt(2) * np.array([[cosalpha, sinalpha], [-sinalpha, cosalpha]])
+    result = np.einsum("ijt,jt->it", demod_matrix, np.stack([modulated_i, modulated_q]))
+    return np.mean(result, axis=1)
```

Comment:
If you're then taking a mean you could directly sum immediately.

It's not much more abstract than the current `einsum`
```suggestion
    return np.einsum("ijt,jt->i", demod_matrix, np.stack([modulated_i, modulated_q])) / len(time)
```

(To me it's even slightly more readable, because I understand better the disappearing `t` than the `axis=1`...)